doc/create-a-custom-rule.md

```diff
@@ -0,0 +1,242 @@
+
+![a lovely dog](lovely-dog.png)
+```
+
+At this point, we have a working `remark` configuration and a markdown file in the project.
+
+If we run `yarn run lint` we should expect to see in our terminal:
+
+```bash
+$ doc.md: no issues found
+```
+
+All good, the file has been processed, and because we haven't specified any plugins nor lint rule, no issues are found.
+
+[Back to Top](#table-of-contents)
+
+## The no-invalid-gif rule
+
+Let's imagine we want to write a rule that checks whether a `.gif` file is used within an image.
+
+Given the content of our `doc.md` file declared above, we would expect an *error* or *warning* pointing to:
+
+```md
+![a funny cat](funny-cat.gif)
+```
+
+Because the file extension `.gif` in the image tag, violates our rule.
+
+[Back to Top](#table-of-contents)
+
+## Create the custom rule
+
+Let's create a new folder `rules` under the root directory, where we will place all of our custom rules, and create a new file in it named `no-gif-allowed.js`.
+
+```bash
+$ mkdir rules
+$ cd rules
+$ touch no-gif-allowed.js
+$ cd .. # return to project root
+```
+
+*Remember*: the name of the folder and files, and where to place them within your project, is entirely up to you.
+
+In `./rules/no-gif-allowed.js`, let's import the `unified-lint-rule`.
+We then export the result of calling `rule` by providing the *namespace and rule name* (`remark-lint:no-gif-allowed`) as the first argument, and our implementation of the rule (`noGifAllowed`) as the second argument.
+
+```js
+// rules/no-gif-allowed.js
+
+var rule = require("unified-lint-rule");
+function noGifAllowed(tree, file, options) {
+  // rule implementation
+}
+module.exports = rule("remark-lint:no-gif-allowed", noGifAllowed);
+```
+
+Let's say you want all your custom rules to be defined as part of your project namespace. If your project was named `my-project`, then you can export your rule as:
+
+```js
+module.exports = rule("my-project-name:no-gif-allowed", noGifAllowed);
+// or
+module.exports = rule("my-npm-published-package:no-gif-allowed", noGifAllowed);
+```
+
+This can help you when wanting to create a group of rules under the same *label* or *namespace*.
+
+[Back to Top](#table-of-contents)
+
+## Rule arguments
+
+Your rule function will receive three arguments.
+
+```js
+function noGifAllowed(tree, file, options) {}
+```
+
+*   `tree` (*required*): a [mdast](https://github.com/syntax-tree/mdast).
+*   `file` (*required*): a [virtual file format](https://github.com/vfile/vfile).
+*   `options` (*optional*): additional information passed to the rule by the remark plugins definition.
+
+[Back to Top](#table-of-contents)
+
+## Rule implementation
+
+Because we will be inspecting a [mdast](https://github.com/syntax-tree/mdast), which is a markdown abstract syntax tree built upon [unist](https://github.com/syntax-tree/unist), we can take advantage of the many existing [unist utilities](https://github.com/syntax-tree/unist#utilities) to inspect our tree's nodes.
```

Comment:
```suggestion
Because we will be inspecting [mdast](https://github.com/syntax-tree/mdast), which is a markdown abstract syntax tree built upon [unist](https://github.com/syntax-tree/unist), we can take advantage of the many existing [unist utilities](https://github.com/syntax-tree/unist#utilities) to inspect our tree’s nodes.
```